### DISCUSSION TITLE
Event data storage requirements question

### DISCUSSION THREAD

**Discussion by michaeltoohig** [plausible/analytics#588] [Self-Hosted Support] [answered]
Good day and thanks for the great software.

I'm currently using Plausible for my personal webapp which basically has 3 users yet the data storage seems excessive.

```
68M     ./plausible-selfhosted_db-data
8.0K    ./plausible-selfhosted_geoip
11G     ./plausible-selfhosted_event-data
```

Above are the docker volumes for basically 2 months of usage where on average I have 2 visits per day. Is this somehow imply I have 11G of actual event data about my site or has the docker volume just reserved 11G of my disk space for future use?

Preferably I'd like to have as little disk space usage up front as possible since I'm just running my personal project on a $5/month VM with only a couple dozen GBs of storage available to me.

**Comment by ukutaht** [comment, marked as answer]
Hey @michaeltoohig

I'm sorry I can't give you a very useful official answer to this. We run Clickhouse on a large separate server and without docker, no issues with storage size. It's possible this could be due to some logging tables. These links can be helpful:

https://theorangeone.net/posts/calming-down-clickhouse/ cc @RealOrangeOne 
https://github.com/plausible/analytics/issues/301

**Reply by michaeltoohig** [reply]
This is the issue. I ran the command to reclaim clickhouse storage from the excessive logging and got back 50% of my disk space.

```
clickhouse-client -q "SELECT name FROM system.tables WHERE name LIKE '%log%';" | xargs -I{} clickhouse-client -q "TRUNCATE TABLE system.{};"
```

I see the latest self hosted repo has been updated with the other configuration suggestions from @RealOrangeOne article so I'll give the new repo updates a try and I'm confident it should be okay now.

Thanks!

**Reply by thepra** [reply]
How to apply this to an already running docker instance?
edit: Nevermind, here're quick commands:
docker container ls
find the CONTAINER_ID in the list of containers and:
docker exec -it CONTAINER_ID bash

**Reply by thepra** [reply]
Mmmm, from within the docker container of clickhouse how am I supposed to apply those settings from the blog once I updated the configuration?

**Reply by RealOrangeOne** [reply]
You don't apply it to a running clickhouse container. You pass the files into the right place, and then restart it. Clickhouse needs to be restarted to read the new settings. Once that's restarted, you can run the above `clickhouse-client` command to reclaim the data.

**Reply by thepra** [reply]
is there a guide on how to do it correctly? step-by-step?

**Reply by RealOrangeOne** [reply]
Yes, I wrote it. See the one linked above

**Reply by adamu** [reply]
> ```
> clickhouse-client -q "SELECT name FROM system.tables WHERE name LIKE '%log%';" | xargs -I{} clickhouse-client -q "TRUNCATE TABLE system.{};"
> ```

Thank you so much for this. Ran this and my volume went from 33GB to 6MB ☺️ 💆🏼‍♂️ 

Have now mounted the [config](https://github.com/plausible/hosting/tree/master/clickhouse) [files](https://github.com/plausible/hosting/blob/a6fe8d13df3566f7bb3625a1a524f5498d56693d/docker-compose.yml#L21-L22) to disable the logging going forward.

**Comment by RealOrangeOne** [comment]
Yes this definitely looks like the issue of clickhouse writing too much data to disk. Plausible itself doesn't actually write much, but by default Clickhouse keeps a log on disk of every query executed, which is a *lot* of data.

My article above talks about the issue and how to fix it, which you can also see at https://github.com/plausible/hosting/pull/13

**Reply by michaeltoohig** [reply]
Thanks for the article and figuring how to calm down Clickhouse.

**Reply by iwootten** [reply]
Ran this command and my 18mo event-data volume went from 12GB -> 658kB!
